TM1py/Services/GitService.py

```diff
@@ -0,0 +1,275 @@
+            body['Passphrase'] = passphrase
+        
+        for key, value in kwargs.items():
+            body[key] = value
+
+        body_json = json.dumps(body)
+        response = self._rest.POST(url=url, data=body_json)
+
+        return self.from_json(response.json())
+
+    def git_push(self, message: str, author: str, email: str, branch: str = None, new_branch: str = None, force: bool = None, username: str = None,
+                    password: str = None, public_key: str = None, private_key: str = None, passphrase: str = None, execute: bool = None,
+                    **kwargs) -> Response:
+        """ Creates a gitpush plan, returns response
+        :param message: Commit message
+        :param author: Name of commit author
+        :param email: Email of commit author
+        :param branch: The branch which last commit will be used as parent commit for new branch. Must be empty if GIT repo is empty
+        :param new_branch: If specified, creates a new branch and pushes the commit onto it. If not specified, pushes to the branch specified in "Branch"
+        :param force: A flag passed in for evaluating preconditions
+        :param username: GIT username
+        :param password: GIT password
+        :param public_key: SSH public key, available from PAA V2.0.9.4
+        :param private_key: SSH private key, available from PAA V2.0.9.4
+        :param passphrase: Passphrase for decrypting private key, if set
+        :param execute: Executes the plan right away if True
+
+        """
+        url = "/api/v1/GitPush"
+        body = {}
+
+        if message is not None:
+            body['Message'] = message
+        if author is not None:
+            body['Author'] = author
+        if email is not None:
+            body['Email'] = email
+        if branch is not None:
+            body['Branch'] = branch
+        if new_branch is not None:
+            body['NewBranch'] = new_branch
+        if force is not None:
+            body['Force'] = force
+        if username is not None:
+            body['Username'] = username
+        if password is not None:
+            body['Password'] = password
+        if public_key is not None:
+            body['PublicKey'] = public_key
+        if private_key is not None:
+            body['PrivateKey'] = private_key
+        if passphrase is not None:
+            body['Passphrase'] = passphrase
+
+        for key, value in kwargs.items():
+            body[key] = value
+
+        body_json = json.dumps(body)
+        response = self._rest.POST(url=url, data=body_json)
+
+        if execute:
+            plan_id = json.loads(response.content).get('ID')
+            self.git_execute_plan(plan_id=plan_id)
+
+        return response
+
+    def git_pull(self, branch: str, force: bool = None, execute: bool = None, username: str = None, password: str = None, 
+                public_key: str = None, private_key: str = None, passphrase: str = None, **kwargs) -> Response:
```

Comment:
`force` is of type `bool` then the default value should be `False`

Reply:
I think it's kinda valid. If `force` is not passed to the function, it will not be included in the body of the request and TM1 would use the default.

Do you think the default behavior should be controlled in TM1py instead through a default value for the `force` argument?

Reply:
IMO having `False` there will make the code look good but would leave it to you to decide.

Reply:
OK